source/extensions/filters/http/ratelimit/ratelimit.h

```diff
@@ -80,6 +83,8 @@ class FilterConfig {
   const bool failure_mode_deny_;
   const absl::optional<Grpc::Status::GrpcStatus> rate_limited_grpc_status_;
   Http::Context& http_context_;
+  Stats::StatNamePool stat_name_pool_;
```

Comment:
Any reason to not just embed this inside the `StatNames` structure on the next line? It seems like that would be less magical?